commit 3df87fa56f208d5ed3122017f49ea2f18be98dac
Author: anhtu <anhtu2687@gmail.com>
Date:   Tue Sep 18 14:44:26 2012 +0700

    JSR286 finished

diff --git a/tuna/Jsr268EventListenerPortlet/.classpath b/tuna/Jsr268EventListenerPortlet/.classpath
new file mode 100644
index 0000000..70a3874
--- /dev/null
+++ b/tuna/Jsr268EventListenerPortlet/.classpath
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<classpath>
+	<classpathentry kind="src" output="target/classes" path="src/main/java"/>
+	<classpathentry excluding="**" kind="src" output="target/classes" path="src/main/resources"/>
+	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/J2SE-1.5"/>
+	<classpathentry kind="con" path="org.maven.ide.eclipse.MAVEN2_CLASSPATH_CONTAINER"/>
+	<classpathentry kind="output" path="target/classes"/>
+</classpath>
diff --git a/tuna/Jsr268EventListenerPortlet/.project b/tuna/Jsr268EventListenerPortlet/.project
new file mode 100644
index 0000000..00abefa
--- /dev/null
+++ b/tuna/Jsr268EventListenerPortlet/.project
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<projectDescription>
+	<name>Jsr268EventListenerPortlet</name>
+	<comment>NO_M2ECLIPSE_SUPPORT: Project files created with the maven-eclipse-plugin are not supported in M2Eclipse.</comment>
+	<projects>
+	</projects>
+	<buildSpec>
+		<buildCommand>
+			<name>org.eclipse.jdt.core.javabuilder</name>
+			<arguments>
+			</arguments>
+		</buildCommand>
+		<buildCommand>
+			<name>org.maven.ide.eclipse.maven2Builder</name>
+			<arguments>
+			</arguments>
+		</buildCommand>
+	</buildSpec>
+	<natures>
+		<nature>org.maven.ide.eclipse.maven2Nature</nature>
+		<nature>org.eclipse.jdt.core.javanature</nature>
+	</natures>
+</projectDescription>
diff --git a/tuna/Jsr268EventListenerPortlet/.settings/org.eclipse.jdt.core.prefs b/tuna/Jsr268EventListenerPortlet/.settings/org.eclipse.jdt.core.prefs
new file mode 100644
index 0000000..abec6ca
--- /dev/null
+++ b/tuna/Jsr268EventListenerPortlet/.settings/org.eclipse.jdt.core.prefs
@@ -0,0 +1,5 @@
+eclipse.preferences.version=1
+org.eclipse.jdt.core.compiler.codegen.targetPlatform=1.5
+org.eclipse.jdt.core.compiler.compliance=1.5
+org.eclipse.jdt.core.compiler.problem.forbiddenReference=warning
+org.eclipse.jdt.core.compiler.source=1.5
diff --git a/tuna/Jsr268EventListenerPortlet/.settings/org.maven.ide.eclipse.prefs b/tuna/Jsr268EventListenerPortlet/.settings/org.maven.ide.eclipse.prefs
new file mode 100644
index 0000000..cfdba58
--- /dev/null
+++ b/tuna/Jsr268EventListenerPortlet/.settings/org.maven.ide.eclipse.prefs
@@ -0,0 +1,7 @@
+activeProfiles=
+eclipse.preferences.version=1
+fullBuildGoals=process-test-resources
+resolveWorkspaceProjects=true
+resourceFilterGoals=process-resources resources\:testResources
+skipCompilerPlugin=true
+version=1
diff --git a/tuna/Jsr268EventListenerPortlet/pom.xml b/tuna/Jsr268EventListenerPortlet/pom.xml
new file mode 100644
index 0000000..c2f856b
--- /dev/null
+++ b/tuna/Jsr268EventListenerPortlet/pom.xml
@@ -0,0 +1,45 @@
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+  <groupId>org.exoplatform.cs</groupId>
+  <artifactId>Jsr268EventListenerPortlet</artifactId>
+  <packaging>war</packaging>
+  <version>1.0-SNAPSHOT</version>
+  <name>Maven Portlet Archetype</name>
+  <url>http://maven.apache.org</url>
+  <profiles>
+      <profile>
+          <id>jetspeed</id>
+          <build>
+              <plugins>
+                  <plugin>
+                      <groupId>org.apache.maven.plugins</groupId>
+                      <artifactId>maven-war-plugin</artifactId>
+                      <version>2.0-beta-2</version>
+                      <configuration>
+                          <webXml>src/main/jetspeed/web.xml</webXml>
+                      </configuration>
+                  </plugin>
+              </plugins>
+          </build>
+      </profile>
+  </profiles>
+  <dependencies>
+      <dependency>
+          <groupId>javax.portlet</groupId>
+          <artifactId>portlet-api</artifactId>
+          <version>2.0</version>
+          <scope>provided</scope>
+      </dependency>
+      <dependency>
+          <groupId>jstl</groupId>
+          <artifactId>jstl</artifactId>
+          <version>1.0.2</version>
+      </dependency>
+      <dependency>
+          <groupId>taglibs</groupId>
+          <artifactId>standard</artifactId>
+          <version>1.0.4</version>
+      </dependency>
+  </dependencies>
+</project>
diff --git a/tuna/Jsr268EventListenerPortlet/src/main/java/org/exoplatform/cs/ContactInfoBean.java b/tuna/Jsr268EventListenerPortlet/src/main/java/org/exoplatform/cs/ContactInfoBean.java
new file mode 100644
index 0000000..daa5bc8
--- /dev/null
+++ b/tuna/Jsr268EventListenerPortlet/src/main/java/org/exoplatform/cs/ContactInfoBean.java
@@ -0,0 +1,32 @@
+package org.exoplatform.cs;
+ 
+/**
+ * this class will be used to display a contact information.
+ * In our example, it will be created when the contactInfo event will be received
+ * @author
+ *
+ */
+public class ContactInfoBean {
+    private String name = null;
+    private String email = null;
+ 
+    public ContactInfoBean() {
+        super();
+    }
+ 
+    public String getEmail() {
+        return email;
+    }
+ 
+    public void setEmail(String email) {
+        this.email = email;
+    }
+ 
+    public String getName() {
+        return name;
+    }
+ 
+    public void setName(String name) {
+        this.name = name;
+    }
+}
\ No newline at end of file
diff --git a/tuna/Jsr268EventListenerPortlet/src/main/java/org/exoplatform/cs/Jsr286EventListenerPortlet.java b/tuna/Jsr268EventListenerPortlet/src/main/java/org/exoplatform/cs/Jsr286EventListenerPortlet.java
new file mode 100644
index 0000000..d4a3e0c
--- /dev/null
+++ b/tuna/Jsr268EventListenerPortlet/src/main/java/org/exoplatform/cs/Jsr286EventListenerPortlet.java
@@ -0,0 +1,62 @@
+package org.exoplatform.cs;
+
+/*
+ * Copyright (C) 2003-2012 eXo Platform SAS.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+
+import java.io.IOException;
+import javax.portlet.EventRequest;
+import javax.portlet.EventResponse;
+import javax.portlet.GenericPortlet;
+import javax.portlet.PortletException;
+import javax.portlet.ProcessEvent;
+import javax.portlet.RenderMode;
+import javax.portlet.RenderRequest;
+import javax.portlet.RenderResponse;
+
+/**
+ * Created by The eXo Platform SAS
+ * Author : eXoPlatform
+ *          tuna@exoplatform.com
+ * Sept 18, 2012  
+ */
+
+public class Jsr286EventListenerPortlet extends GenericPortlet {
+	private ContactInfoBean infoBean = new ContactInfoBean();
+	
+	@RenderMode(name="view")
+	public void viewNormal(RenderRequest request, RenderResponse response) throws PortletException, IOException {
+		request.setAttribute("info", infoBean);
+		getPortletContext().getRequestDispatcher("/xhtml/view.jsp").forward(request, response);
+	}
+
+	/** This method processes received events with the following QName
+	 */
+	@ProcessEvent(qname="{http://exoplatform.com/events}contactInfo")
+	public void processContactInfoEvent(EventRequest request, EventResponse response) throws PortletException, IOException {
+
+		String contactInfo = (String) request.getEvent().getValue();
+
+		try {
+			infoBean.setName(contactInfo.split(",")[0]);
+			infoBean.setEmail(contactInfo.split(",")[1]);
+		} catch (Exception e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
+		}
+	} 
+}
\ No newline at end of file
diff --git a/tuna/Jsr268EventListenerPortlet/src/main/jetspeed/web.xml b/tuna/Jsr268EventListenerPortlet/src/main/jetspeed/web.xml
new file mode 100644
index 0000000..8f5115e
--- /dev/null
+++ b/tuna/Jsr268EventListenerPortlet/src/main/jetspeed/web.xml
@@ -0,0 +1,36 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<!DOCTYPE web-app PUBLIC "-//Sun Microsystems, Inc.//DTD Web Application 2.3//EN"
+                         "http://java.sun.com/dtd/web-app_2_3.dtd">
+<web-app>
+    <display-name>Jsr268EventListenerPortlet Wrapper</display-name>
+
+    <description>Generated Portlet Wrapper</description>
+
+    <servlet>
+        <servlet-name>Jsr268EventListenerPortlet</servlet-name>
+        <display-name>Jsr268EventListenerPortlet Portlet</display-name>
+        <description>Portlet for Jetspeed Fusion</description>
+        <servlet-class>org.apache.jetspeed.container.JetspeedContainerServlet</servlet-class>
+
+        <init-param>
+            <param-name>registerAtInit</param-name>
+            <param-value>1</param-value>
+        </init-param>
+
+        <init-param>
+            <param-name>portletApplication</param-name>
+            <param-value>Jsr268EventListenerPortlet</param-value>
+        </init-param>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>JetspeedContainer</servlet-name>
+        <url-pattern>/container/*</url-pattern>
+    </servlet-mapping>
+
+    <taglib>
+        <taglib-uri>http://java.sun.com/portlet</taglib-uri>
+        <taglib-location>/WEB-INF/tld/portlet.tld</taglib-location>
+    </taglib>
+</web-app>
diff --git a/tuna/Jsr268EventListenerPortlet/src/main/webapp/WEB-INF/portlet.xml b/tuna/Jsr268EventListenerPortlet/src/main/webapp/WEB-INF/portlet.xml
new file mode 100644
index 0000000..5ce1e11
--- /dev/null
+++ b/tuna/Jsr268EventListenerPortlet/src/main/webapp/WEB-INF/portlet.xml
@@ -0,0 +1,37 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<portlet-app version="1.0" xmlns="http://java.sun.com/xml/ns/portlet/portlet-app_2_0.xsd">
+ 
+<portlet>
+    <portlet-name>Jsr286EventListenerPortlet</portlet-name>
+    <portlet-class>org.exoplatform.cs.Jsr286EventListenerPortlet</portlet-class>
+	    
+    <init-param>
+		<description>view jsp</description>
+	        <name>jspView</name>
+		<value>/xhtml/view.jsp</value>
+	</init-param>
+    
+    <supports>
+        <mime-type>text/html</mime-type>
+        <portlet-mode>VIEW</portlet-mode>
+    </supports>
+    <supported-locale>en</supported-locale>
+ 
+	<portlet-info>
+        <title>Jsr286 Event Listener Portlet</title>
+        <short-title>Jsr286EventListenerPortlet</short-title>
+        <keywords>Jsr286EventListenerPortlet</keywords>
+    </portlet-info>
+    <supported-processing-event>
+        <qname xmlns:x="http://exoplatform.com/events">x:contactInfo</qname>
+    </supported-processing-event>
+</portlet>
+ 
+    <!-- <default-namespace>ns.Jsr286EventListenerPortlet</default-namespace> -->
+ 
+    <event-definition>
+        <qname xmlns:x="http://exoplatform.com/events">x:contactInfo</qname>
+        <value-type>java.lang.String</value-type>
+    </event-definition>
+ 
+</portlet-app>
\ No newline at end of file
diff --git a/tuna/Jsr268EventListenerPortlet/src/main/webapp/WEB-INF/tld/portlet.tld b/tuna/Jsr268EventListenerPortlet/src/main/webapp/WEB-INF/tld/portlet.tld
new file mode 100644
index 0000000..b4af8ea
--- /dev/null
+++ b/tuna/Jsr268EventListenerPortlet/src/main/webapp/WEB-INF/tld/portlet.tld
@@ -0,0 +1,104 @@
+<?xml version="1.0" encoding="ISO-8859-1" ?>
+<!DOCTYPE taglib PUBLIC
+  "-//Sun Microsystems, Inc.//DTD JSP Tag Library 1.1//EN"
+  "http://java.sun.com/j2ee/dtds/web-jsptaglibrary_1_1.dtd">
+<!-- 
+Copyright 2004 The Apache Software Foundation
+Licensed  under the  Apache License,  Version 2.0  (the "License");
+you may not use  this file  except in  compliance with the License.
+You may obtain a copy of the License at 
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed  under the  License is distributed on an "AS IS" BASIS,
+WITHOUT  WARRANTIES OR CONDITIONS  OF ANY KIND, either  express  or
+implied.
+
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+<taglib>
+    <tlibversion>1.0</tlibversion>
+    <jspversion>1.1</jspversion>
+    <shortname>portlet</shortname>
+    <uri>http://java.sun.com/portlet</uri>
+    <tag>
+        <name>defineObjects</name>
+        <tagclass>org.apache.pluto.tags.DefineObjectsTag</tagclass>
+        <teiclass>org.apache.pluto.tags.DefineObjectsTag$TEI</teiclass>
+        <bodycontent>empty</bodycontent>
+    </tag>
+    <tag>
+        <name>param</name>
+        <tagclass>org.apache.pluto.tags.ParamTag</tagclass>
+        <bodycontent>empty</bodycontent>
+        <attribute>
+            <name>name</name>
+            <required>true</required>
+            <rtexprvalue>true</rtexprvalue>
+        </attribute>
+        <attribute>
+            <name>value</name>
+            <required>true</required>
+            <rtexprvalue>true</rtexprvalue>
+        </attribute>
+    </tag>
+    <tag>
+        <name>actionURL</name>
+        <tagclass>org.apache.pluto.tags.ActionURLTag</tagclass>
+        <teiclass>org.apache.pluto.tags.BasicURLTag$TEI</teiclass>
+        <bodycontent>JSP</bodycontent>
+        <attribute>
+            <name>windowState</name>
+            <required>false</required>
+            <rtexprvalue>true</rtexprvalue>
+        </attribute>
+        <attribute>
+            <name>portletMode</name>
+            <required>false</required>
+            <rtexprvalue>true</rtexprvalue>
+        </attribute>
+        <attribute>
+            <name>secure</name>
+            <required>false</required>
+            <rtexprvalue>true</rtexprvalue>
+        </attribute>
+        <attribute>
+            <name>var</name>
+            <required>false</required>
+            <rtexprvalue>true</rtexprvalue>
+        </attribute>
+    </tag>
+    <tag>
+        <name>renderURL</name>
+        <tagclass>org.apache.pluto.tags.RenderURLTag</tagclass>
+        <teiclass>org.apache.pluto.tags.BasicURLTag$TEI</teiclass>
+        <bodycontent>JSP</bodycontent>
+        <attribute>
+            <name>windowState</name>
+            <required>false</required>
+            <rtexprvalue>true</rtexprvalue>
+        </attribute>
+        <attribute>
+            <name>portletMode</name>
+            <required>false</required>
+            <rtexprvalue>true</rtexprvalue>
+        </attribute>
+        <attribute>
+            <name>secure</name>
+            <required>false</required>
+            <rtexprvalue>true</rtexprvalue>
+        </attribute>
+        <attribute>
+            <name>var</name>
+            <required>false</required>
+            <rtexprvalue>true</rtexprvalue>
+        </attribute>
+    </tag>
+    <tag>
+        <name>namespace</name>
+        <tagclass>org.apache.pluto.tags.NamespaceTag</tagclass>
+        <bodycontent>empty</bodycontent>
+    </tag>
+</taglib>
diff --git a/tuna/Jsr268EventListenerPortlet/src/main/webapp/WEB-INF/web.xml b/tuna/Jsr268EventListenerPortlet/src/main/webapp/WEB-INF/web.xml
new file mode 100644
index 0000000..f2d8101
--- /dev/null
+++ b/tuna/Jsr268EventListenerPortlet/src/main/webapp/WEB-INF/web.xml
@@ -0,0 +1,24 @@
+<?xml version="1.0" encoding="UTF-8"?>
+ 
+<web-app id="Jsr286EventListenerPortlet" version="2.4" xmlns="http://java.sun.com/xml/ns/j2ee" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://java.sun.com/xml/ns/j2ee http://java.sun.com/xml/ns/j2ee/web-app_2_4.xsd">
+    <display-name>Jsr286EventListenerPortlet</display-name>
+ 
+    <listener>
+        <listener-class>org.exoplatform.services.portletcontainer.impl.servlet.PortletApplicationListener</listener-class>
+    </listener>
+ 
+    <servlet>
+        <servlet-name>PortletWrapper</servlet-name>
+        <servlet-class>org.exoplatform.services.portletcontainer.impl.servlet.ServletWrapper</servlet-class>
+    </servlet>
+    <servlet-mapping>
+        <servlet-name>PortletWrapper</servlet-name>
+        <url-pattern>/PortletWrapper</url-pattern>
+    </servlet-mapping>
+ 
+    <taglib>
+        <taglib-uri></taglib-uri>
+        <taglib-location>/WEB-INF/tld/portlet.tld</taglib-location>
+    </taglib>
+ 
+</web-app>
\ No newline at end of file
diff --git a/tuna/Jsr268EventListenerPortlet/src/main/webapp/xhtml/view.jsp b/tuna/Jsr268EventListenerPortlet/src/main/webapp/xhtml/view.jsp
new file mode 100644
index 0000000..56757c9
--- /dev/null
+++ b/tuna/Jsr268EventListenerPortlet/src/main/webapp/xhtml/view.jsp
@@ -0,0 +1,30 @@
+<%@page import="org.exoplatform.cs.ContactInfoBean"%>
+
+<%!ContactInfoBean contactInfo = null;%>
+<%
+//retrieve the object from the session
+contactInfo = (ContactInfoBean) request.getAttribute("info");
+ 
+if ((contactInfo.getName() == null) && (contactInfo.getEmail() ==null)) {
+%>
+	<div><h4>No contact information.</h4></div>
+    
+<% } else {
+%>
+	<div>
+        <table>
+        <tbody>
+            <tr>
+                <th>Name:</th>
+                <td><%=contactInfo.getName() %></td>
+            </tr>
+            <tr>
+                <th>Email:</th>
+                <td><%=contactInfo.getEmail() %></td>
+            </tr>
+        </tbody>
+        </table>
+    </div>
+<%
+}
+%>
\ No newline at end of file
diff --git a/tuna/Jsr268EventListenerPortlet/target/.DS_Store b/tuna/Jsr268EventListenerPortlet/target/.DS_Store
new file mode 100644
index 0000000..5008ddf
Binary files /dev/null and b/tuna/Jsr268EventListenerPortlet/target/.DS_Store differ
diff --git a/tuna/Jsr268EventListenerPortlet/target/Jsr268EventListenerPortlet-1.0-SNAPSHOT/WEB-INF/portlet.xml b/tuna/Jsr268EventListenerPortlet/target/Jsr268EventListenerPortlet-1.0-SNAPSHOT/WEB-INF/portlet.xml
new file mode 100644
index 0000000..5ce1e11
--- /dev/null
+++ b/tuna/Jsr268EventListenerPortlet/target/Jsr268EventListenerPortlet-1.0-SNAPSHOT/WEB-INF/portlet.xml
@@ -0,0 +1,37 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<portlet-app version="1.0" xmlns="http://java.sun.com/xml/ns/portlet/portlet-app_2_0.xsd">
+ 
+<portlet>
+    <portlet-name>Jsr286EventListenerPortlet</portlet-name>
+    <portlet-class>org.exoplatform.cs.Jsr286EventListenerPortlet</portlet-class>
+	    
+    <init-param>
+		<description>view jsp</description>
+	        <name>jspView</name>
+		<value>/xhtml/view.jsp</value>
+	</init-param>
+    
+    <supports>
+        <mime-type>text/html</mime-type>
+        <portlet-mode>VIEW</portlet-mode>
+    </supports>
+    <supported-locale>en</supported-locale>
+ 
+	<portlet-info>
+        <title>Jsr286 Event Listener Portlet</title>
+        <short-title>Jsr286EventListenerPortlet</short-title>
+        <keywords>Jsr286EventListenerPortlet</keywords>
+    </portlet-info>
+    <supported-processing-event>
+        <qname xmlns:x="http://exoplatform.com/events">x:contactInfo</qname>
+    </supported-processing-event>
+</portlet>
+ 
+    <!-- <default-namespace>ns.Jsr286EventListenerPortlet</default-namespace> -->
+ 
+    <event-definition>
+        <qname xmlns:x="http://exoplatform.com/events">x:contactInfo</qname>
+        <value-type>java.lang.String</value-type>
+    </event-definition>
+ 
+</portlet-app>
\ No newline at end of file
diff --git a/tuna/Jsr268EventListenerPortlet/target/Jsr268EventListenerPortlet-1.0-SNAPSHOT/WEB-INF/tld/portlet.tld b/tuna/Jsr268EventListenerPortlet/target/Jsr268EventListenerPortlet-1.0-SNAPSHOT/WEB-INF/tld/portlet.tld
new file mode 100644
index 0000000..b4af8ea
--- /dev/null
+++ b/tuna/Jsr268EventListenerPortlet/target/Jsr268EventListenerPortlet-1.0-SNAPSHOT/WEB-INF/tld/portlet.tld
@@ -0,0 +1,104 @@
+<?xml version="1.0" encoding="ISO-8859-1" ?>
+<!DOCTYPE taglib PUBLIC
+  "-//Sun Microsystems, Inc.//DTD JSP Tag Library 1.1//EN"
+  "http://java.sun.com/j2ee/dtds/web-jsptaglibrary_1_1.dtd">
+<!-- 
+Copyright 2004 The Apache Software Foundation
+Licensed  under the  Apache License,  Version 2.0  (the "License");
+you may not use  this file  except in  compliance with the License.
+You may obtain a copy of the License at 
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed  under the  License is distributed on an "AS IS" BASIS,
+WITHOUT  WARRANTIES OR CONDITIONS  OF ANY KIND, either  express  or
+implied.
+
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+<taglib>
+    <tlibversion>1.0</tlibversion>
+    <jspversion>1.1</jspversion>
+    <shortname>portlet</shortname>
+    <uri>http://java.sun.com/portlet</uri>
+    <tag>
+        <name>defineObjects</name>
+        <tagclass>org.apache.pluto.tags.DefineObjectsTag</tagclass>
+        <teiclass>org.apache.pluto.tags.DefineObjectsTag$TEI</teiclass>
+        <bodycontent>empty</bodycontent>
+    </tag>
+    <tag>
+        <name>param</name>
+        <tagclass>org.apache.pluto.tags.ParamTag</tagclass>
+        <bodycontent>empty</bodycontent>
+        <attribute>
+            <name>name</name>
+            <required>true</required>
+            <rtexprvalue>true</rtexprvalue>
+        </attribute>
+        <attribute>
+            <name>value</name>
+            <required>true</required>
+            <rtexprvalue>true</rtexprvalue>
+        </attribute>
+    </tag>
+    <tag>
+        <name>actionURL</name>
+        <tagclass>org.apache.pluto.tags.ActionURLTag</tagclass>
+        <teiclass>org.apache.pluto.tags.BasicURLTag$TEI</teiclass>
+        <bodycontent>JSP</bodycontent>
+        <attribute>
+            <name>windowState</name>
+            <required>false</required>
+            <rtexprvalue>true</rtexprvalue>
+        </attribute>
+        <attribute>
+            <name>portletMode</name>
+            <required>false</required>
+            <rtexprvalue>true</rtexprvalue>
+        </attribute>
+        <attribute>
+            <name>secure</name>
+            <required>false</required>
+            <rtexprvalue>true</rtexprvalue>
+        </attribute>
+        <attribute>
+            <name>var</name>
+            <required>false</required>
+            <rtexprvalue>true</rtexprvalue>
+        </attribute>
+    </tag>
+    <tag>
+        <name>renderURL</name>
+        <tagclass>org.apache.pluto.tags.RenderURLTag</tagclass>
+        <teiclass>org.apache.pluto.tags.BasicURLTag$TEI</teiclass>
+        <bodycontent>JSP</bodycontent>
+        <attribute>
+            <name>windowState</name>
+            <required>false</required>
+            <rtexprvalue>true</rtexprvalue>
+        </attribute>
+        <attribute>
+            <name>portletMode</name>
+            <required>false</required>
+            <rtexprvalue>true</rtexprvalue>
+        </attribute>
+        <attribute>
+            <name>secure</name>
+            <required>false</required>
+            <rtexprvalue>true</rtexprvalue>
+        </attribute>
+        <attribute>
+            <name>var</name>
+            <required>false</required>
+            <rtexprvalue>true</rtexprvalue>
+        </attribute>
+    </tag>
+    <tag>
+        <name>namespace</name>
+        <tagclass>org.apache.pluto.tags.NamespaceTag</tagclass>
+        <bodycontent>empty</bodycontent>
+    </tag>
+</taglib>
diff --git a/tuna/Jsr268EventListenerPortlet/target/Jsr268EventListenerPortlet-1.0-SNAPSHOT/WEB-INF/web.xml b/tuna/Jsr268EventListenerPortlet/target/Jsr268EventListenerPortlet-1.0-SNAPSHOT/WEB-INF/web.xml
new file mode 100644
index 0000000..f2d8101
--- /dev/null
+++ b/tuna/Jsr268EventListenerPortlet/target/Jsr268EventListenerPortlet-1.0-SNAPSHOT/WEB-INF/web.xml
@@ -0,0 +1,24 @@
+<?xml version="1.0" encoding="UTF-8"?>
+ 
+<web-app id="Jsr286EventListenerPortlet" version="2.4" xmlns="http://java.sun.com/xml/ns/j2ee" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://java.sun.com/xml/ns/j2ee http://java.sun.com/xml/ns/j2ee/web-app_2_4.xsd">
+    <display-name>Jsr286EventListenerPortlet</display-name>
+ 
+    <listener>
+        <listener-class>org.exoplatform.services.portletcontainer.impl.servlet.PortletApplicationListener</listener-class>
+    </listener>
+ 
+    <servlet>
+        <servlet-name>PortletWrapper</servlet-name>
+        <servlet-class>org.exoplatform.services.portletcontainer.impl.servlet.ServletWrapper</servlet-class>
+    </servlet>
+    <servlet-mapping>
+        <servlet-name>PortletWrapper</servlet-name>
+        <url-pattern>/PortletWrapper</url-pattern>
+    </servlet-mapping>
+ 
+    <taglib>
+        <taglib-uri></taglib-uri>
+        <taglib-location>/WEB-INF/tld/portlet.tld</taglib-location>
+    </taglib>
+ 
+</web-app>
\ No newline at end of file
diff --git a/tuna/Jsr268EventListenerPortlet/target/Jsr268EventListenerPortlet-1.0-SNAPSHOT/xhtml/view.jsp b/tuna/Jsr268EventListenerPortlet/target/Jsr268EventListenerPortlet-1.0-SNAPSHOT/xhtml/view.jsp
new file mode 100644
index 0000000..56757c9
--- /dev/null
+++ b/tuna/Jsr268EventListenerPortlet/target/Jsr268EventListenerPortlet-1.0-SNAPSHOT/xhtml/view.jsp
@@ -0,0 +1,30 @@
+<%@page import="org.exoplatform.cs.ContactInfoBean"%>
+
+<%!ContactInfoBean contactInfo = null;%>
+<%
+//retrieve the object from the session
+contactInfo = (ContactInfoBean) request.getAttribute("info");
+ 
+if ((contactInfo.getName() == null) && (contactInfo.getEmail() ==null)) {
+%>
+	<div><h4>No contact information.</h4></div>
+    
+<% } else {
+%>
+	<div>
+        <table>
+        <tbody>
+            <tr>
+                <th>Name:</th>
+                <td><%=contactInfo.getName() %></td>
+            </tr>
+            <tr>
+                <th>Email:</th>
+                <td><%=contactInfo.getEmail() %></td>
+            </tr>
+        </tbody>
+        </table>
+    </div>
+<%
+}
+%>
\ No newline at end of file
diff --git a/tuna/Jsr268EventListenerPortlet/target/maven-archiver/pom.properties b/tuna/Jsr268EventListenerPortlet/target/maven-archiver/pom.properties
new file mode 100644
index 0000000..3ccc640
--- /dev/null
+++ b/tuna/Jsr268EventListenerPortlet/target/maven-archiver/pom.properties
@@ -0,0 +1,5 @@
+#Generated by Maven
+#Tue Sep 18 07:38:24 GMT+00:07 2012
+version=1.0-SNAPSHOT
+groupId=org.exoplatform.cs
+artifactId=Jsr268EventListenerPortlet
diff --git a/tuna/Jsr268EventPublisherPortlet/pom.xml b/tuna/Jsr268EventPublisherPortlet/pom.xml
index 95d0daf..8630f28 100644
--- a/tuna/Jsr268EventPublisherPortlet/pom.xml
+++ b/tuna/Jsr268EventPublisherPortlet/pom.xml
@@ -26,7 +26,7 @@
   </profiles>
   <dependencies>
       <dependency>
-          <groupId>portlet-api</groupId>
+          <groupId>javax.portlet</groupId>
           <artifactId>portlet-api</artifactId>
           <version>2.0</version>
           <scope>provided</scope>
diff --git a/tuna/Jsr268EventPublisherPortlet/src/main/java/org/exoplatform/cs/Jsr286EventPublisherPortlet.java b/tuna/Jsr268EventPublisherPortlet/src/main/java/org/exoplatform/cs/Jsr286EventPublisherPortlet.java
index 9b11fa4..550a132 100644
--- a/tuna/Jsr268EventPublisherPortlet/src/main/java/org/exoplatform/cs/Jsr286EventPublisherPortlet.java
+++ b/tuna/Jsr268EventPublisherPortlet/src/main/java/org/exoplatform/cs/Jsr286EventPublisherPortlet.java
@@ -1,21 +1,23 @@
 package org.exoplatform.cs;
 
 /*
- * Copyright 2001-2005 The Apache Software Foundation.
+ * Copyright (C) 2003-2012 eXo Platform SAS.
  *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
  *
- *      http://www.apache.org/licenses/LICENSE-2.0
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
  *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
  */
 
+
 import java.io.IOException;
 import javax.portlet.ActionRequest;
 import javax.portlet.ActionResponse;
@@ -26,27 +28,31 @@ import javax.portlet.RenderMode;
 import javax.portlet.RenderRequest;
 import javax.portlet.RenderResponse;
 import javax.xml.namespace.QName;
-// what the fck is that??
+
+/**
+ * Created by The eXo Platform SAS
+ * Author : eXoPlatform
+ *          tuna@exoplatform.com
+ * Sept 18, 2012  
+ */
 
 public class Jsr286EventPublisherPortlet extends GenericPortlet {
 
-    @RenderMode(name="help")
+    @RenderMode(name="help") // when having a help page request click on Help, give back help.jsp -- MAPPING purpose - tested
     protected void doHelp( RenderRequest request, RenderResponse response )
         throws PortletException, IOException {
     	getPortletContext().getRequestDispatcher("/xhtml/help.jsp").include(request, response);
-        //helpView.include( request, response );
     }
     
     @RenderMode(name="view") // this annotation indicates that the render is for viewing purpose
     public void viewNormal( RenderRequest request, RenderResponse response )
         throws PortletException, IOException {
-
+    	
     	getPortletContext().getRequestDispatcher("/xhtml/view.jsp").forward(request, response);
     }
 
     /**
      * This method processes the "savecontact" action.
-     * The form parameters are added to the Hashtable object that will be sent as the event value
      */
     @ProcessAction(name="savecontact")
     public void saveContact(ActionRequest request, ActionResponse response) throws PortletException, IOException {
@@ -54,12 +60,13 @@ public class Jsr286EventPublisherPortlet extends GenericPortlet {
         // Get form parameters
         String name = request.getParameter("name");
         String email = request.getParameter("email");
- 
+        
         //send event
         String contactInfo = name + "," + email;
  
         // Send the event using the appropriate QName
-        response.setEvent(new QName("http:mycompany.com/events", "contactInfo"), contactInfo);
+        response.setEvent(new QName("http://exoplatform.com/events", "contactInfo"), contactInfo);
+        // send contactInfo event to http://exoplatform.com/events, ListenerPortlet will retrieve the event from this
     }
 
 }
diff --git a/tuna/Jsr268EventPublisherPortlet/src/main/webapp/WEB-INF/portlet.xml b/tuna/Jsr268EventPublisherPortlet/src/main/webapp/WEB-INF/portlet.xml
index cd7276c..56bff5e 100644
--- a/tuna/Jsr268EventPublisherPortlet/src/main/webapp/WEB-INF/portlet.xml
+++ b/tuna/Jsr268EventPublisherPortlet/src/main/webapp/WEB-INF/portlet.xml
@@ -5,27 +5,26 @@
 	<portlet-name>Jsr286EventPublisherPortlet</portlet-name>
 	<portlet-class>org.exoplatform.cs.Jsr286EventPublisherPortlet</portlet-class>
 
-	<portlet-info>
-	<title>Jsr286 Event Publisher Portlet</title>
-	<short-title>Jsr286EventPublisherPortlet</short-title>
-	</portlet-info>
-
 	<supports>
 	<mime-type>text/html</mime-type>
 	<portlet-mode>VIEW</portlet-mode>
 	<portlet-mode>HELP</portlet-mode>
 	</supports>
-
 	<supported-locale>en</supported-locale>
 
+	<portlet-info>
+	<title>Jsr286 Event Publisher Portlet</title>
+	<short-title>Jsr286EventPublisherPortlet</short-title>
+	</portlet-info>
+	
 	<supported-publishing-event>
-	<qname xmlns:x="http:mycompany.com/events">x:contactInfo</qname>
+	<qname xmlns:x="http://exoplatform.com/events">x:contactInfo</qname>
 	</supported-publishing-event>
 	<!-- Event -->
 </portlet>
 
 <event-definition>
-	<qname xmlns:x="http:mycompany.com/events">x:contactInfo</qname>
+	<qname xmlns:x="http://exoplatform.com/events">x:contactInfo</qname>
 	<value-type>java.lang.String</value-type>
 </event-definition>
 
diff --git a/tuna/Jsr268EventPublisherPortlet/src/main/webapp/xhtml/help.jsp b/tuna/Jsr268EventPublisherPortlet/src/main/webapp/xhtml/help.jsp
index aa629ef..276b480 100644
--- a/tuna/Jsr268EventPublisherPortlet/src/main/webapp/xhtml/help.jsp
+++ b/tuna/Jsr268EventPublisherPortlet/src/main/webapp/xhtml/help.jsp
@@ -1,3 +1,3 @@
 <%@ taglib uri="http://java.sun.com/jstl/core" prefix="c" %>
- 
+
 <h4>Need help?</h4>
\ No newline at end of file
diff --git a/tuna/Jsr268EventPublisherPortlet/src/main/webapp/xhtml/view.jsp b/tuna/Jsr268EventPublisherPortlet/src/main/webapp/xhtml/view.jsp
index c5ddd8c..8c001f0 100644
--- a/tuna/Jsr268EventPublisherPortlet/src/main/webapp/xhtml/view.jsp
+++ b/tuna/Jsr268EventPublisherPortlet/src/main/webapp/xhtml/view.jsp
@@ -2,14 +2,15 @@
 <%@ taglib uri="http://java.sun.com/jstl/core" prefix="c" %>
 <portlet:defineObjects/>
  
-<div>
-    <p>Use the form below to create a contact. The data is sent over to other portlets into a HashMap.</p>
+ <div>
+    <p>Use the form below to create a contact. The data is sent over to other portlets.</p>
     <%-- The action name must be passed as a param, not as an actionURL attribute. Fix will come with Liferay 5.0.2 --%>
     <form method="post" action="<portlet:actionURL>
     
     <portlet:param name="javax.portlet.action" value="savecontact" />
     </portlet:actionURL>">
-        
+    <%-- define savecontact ACTION, this action is processed by ProcessAction savecontact --%>
+    
     <fieldset>
     <table>
     <tbody>
diff --git a/tuna/Jsr268EventPublisherPortlet/target/.DS_Store b/tuna/Jsr268EventPublisherPortlet/target/.DS_Store
new file mode 100644
index 0000000..5008ddf
Binary files /dev/null and b/tuna/Jsr268EventPublisherPortlet/target/.DS_Store differ
diff --git a/tuna/Jsr268EventPublisherPortlet/target/Jsr268EventPublisherPortlet-1.0-SNAPSHOT/WEB-INF/portlet.xml b/tuna/Jsr268EventPublisherPortlet/target/Jsr268EventPublisherPortlet-1.0-SNAPSHOT/WEB-INF/portlet.xml
new file mode 100644
index 0000000..56bff5e
--- /dev/null
+++ b/tuna/Jsr268EventPublisherPortlet/target/Jsr268EventPublisherPortlet-1.0-SNAPSHOT/WEB-INF/portlet.xml
@@ -0,0 +1,31 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<portlet-app version="1.0" xmlns="http://java.sun.com/xml/ns/portlet/portlet-app_2_0.xsd">
+
+<portlet>
+	<portlet-name>Jsr286EventPublisherPortlet</portlet-name>
+	<portlet-class>org.exoplatform.cs.Jsr286EventPublisherPortlet</portlet-class>
+
+	<supports>
+	<mime-type>text/html</mime-type>
+	<portlet-mode>VIEW</portlet-mode>
+	<portlet-mode>HELP</portlet-mode>
+	</supports>
+	<supported-locale>en</supported-locale>
+
+	<portlet-info>
+	<title>Jsr286 Event Publisher Portlet</title>
+	<short-title>Jsr286EventPublisherPortlet</short-title>
+	</portlet-info>
+	
+	<supported-publishing-event>
+	<qname xmlns:x="http://exoplatform.com/events">x:contactInfo</qname>
+	</supported-publishing-event>
+	<!-- Event -->
+</portlet>
+
+<event-definition>
+	<qname xmlns:x="http://exoplatform.com/events">x:contactInfo</qname>
+	<value-type>java.lang.String</value-type>
+</event-definition>
+
+</portlet-app>
\ No newline at end of file
diff --git a/tuna/Jsr268EventPublisherPortlet/target/Jsr268EventPublisherPortlet-1.0-SNAPSHOT/WEB-INF/tld/portlet.tld b/tuna/Jsr268EventPublisherPortlet/target/Jsr268EventPublisherPortlet-1.0-SNAPSHOT/WEB-INF/tld/portlet.tld
new file mode 100644
index 0000000..b4af8ea
--- /dev/null
+++ b/tuna/Jsr268EventPublisherPortlet/target/Jsr268EventPublisherPortlet-1.0-SNAPSHOT/WEB-INF/tld/portlet.tld
@@ -0,0 +1,104 @@
+<?xml version="1.0" encoding="ISO-8859-1" ?>
+<!DOCTYPE taglib PUBLIC
+  "-//Sun Microsystems, Inc.//DTD JSP Tag Library 1.1//EN"
+  "http://java.sun.com/j2ee/dtds/web-jsptaglibrary_1_1.dtd">
+<!-- 
+Copyright 2004 The Apache Software Foundation
+Licensed  under the  Apache License,  Version 2.0  (the "License");
+you may not use  this file  except in  compliance with the License.
+You may obtain a copy of the License at 
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed  under the  License is distributed on an "AS IS" BASIS,
+WITHOUT  WARRANTIES OR CONDITIONS  OF ANY KIND, either  express  or
+implied.
+
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+<taglib>
+    <tlibversion>1.0</tlibversion>
+    <jspversion>1.1</jspversion>
+    <shortname>portlet</shortname>
+    <uri>http://java.sun.com/portlet</uri>
+    <tag>
+        <name>defineObjects</name>
+        <tagclass>org.apache.pluto.tags.DefineObjectsTag</tagclass>
+        <teiclass>org.apache.pluto.tags.DefineObjectsTag$TEI</teiclass>
+        <bodycontent>empty</bodycontent>
+    </tag>
+    <tag>
+        <name>param</name>
+        <tagclass>org.apache.pluto.tags.ParamTag</tagclass>
+        <bodycontent>empty</bodycontent>
+        <attribute>
+            <name>name</name>
+            <required>true</required>
+            <rtexprvalue>true</rtexprvalue>
+        </attribute>
+        <attribute>
+            <name>value</name>
+            <required>true</required>
+            <rtexprvalue>true</rtexprvalue>
+        </attribute>
+    </tag>
+    <tag>
+        <name>actionURL</name>
+        <tagclass>org.apache.pluto.tags.ActionURLTag</tagclass>
+        <teiclass>org.apache.pluto.tags.BasicURLTag$TEI</teiclass>
+        <bodycontent>JSP</bodycontent>
+        <attribute>
+            <name>windowState</name>
+            <required>false</required>
+            <rtexprvalue>true</rtexprvalue>
+        </attribute>
+        <attribute>
+            <name>portletMode</name>
+            <required>false</required>
+            <rtexprvalue>true</rtexprvalue>
+        </attribute>
+        <attribute>
+            <name>secure</name>
+            <required>false</required>
+            <rtexprvalue>true</rtexprvalue>
+        </attribute>
+        <attribute>
+            <name>var</name>
+            <required>false</required>
+            <rtexprvalue>true</rtexprvalue>
+        </attribute>
+    </tag>
+    <tag>
+        <name>renderURL</name>
+        <tagclass>org.apache.pluto.tags.RenderURLTag</tagclass>
+        <teiclass>org.apache.pluto.tags.BasicURLTag$TEI</teiclass>
+        <bodycontent>JSP</bodycontent>
+        <attribute>
+            <name>windowState</name>
+            <required>false</required>
+            <rtexprvalue>true</rtexprvalue>
+        </attribute>
+        <attribute>
+            <name>portletMode</name>
+            <required>false</required>
+            <rtexprvalue>true</rtexprvalue>
+        </attribute>
+        <attribute>
+            <name>secure</name>
+            <required>false</required>
+            <rtexprvalue>true</rtexprvalue>
+        </attribute>
+        <attribute>
+            <name>var</name>
+            <required>false</required>
+            <rtexprvalue>true</rtexprvalue>
+        </attribute>
+    </tag>
+    <tag>
+        <name>namespace</name>
+        <tagclass>org.apache.pluto.tags.NamespaceTag</tagclass>
+        <bodycontent>empty</bodycontent>
+    </tag>
+</taglib>
diff --git a/tuna/Jsr268EventPublisherPortlet/target/Jsr268EventPublisherPortlet-1.0-SNAPSHOT/WEB-INF/web.xml b/tuna/Jsr268EventPublisherPortlet/target/Jsr268EventPublisherPortlet-1.0-SNAPSHOT/WEB-INF/web.xml
new file mode 100644
index 0000000..8a6eee3
--- /dev/null
+++ b/tuna/Jsr268EventPublisherPortlet/target/Jsr268EventPublisherPortlet-1.0-SNAPSHOT/WEB-INF/web.xml
@@ -0,0 +1,27 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<web-app id="Jsr286EventPublisherPortlet" version="2.4" xmlns="http://java.sun.com/xml/ns/j2ee" 
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" 
+	xsi:schemaLocation="http://java.sun.com/xml/ns/j2ee http://java.sun.com/xml/ns/j2ee/web-app_2_4.xsd">
+
+	<display-name>Jsr286EventPublisherPortlet</display-name>
+	<description>Jsr286 Event Publisher Portlet</description>
+
+	<listener>
+	<listener-class>org.exoplatform.services.portletcontainer.impl.servlet.PortletApplicationListener</listener-class>
+	</listener>
+
+	<servlet>
+	<servlet-name>PortletWrapper</servlet-name>
+	<servlet-class>org.exoplatform.services.portletcontainer.impl.servlet.ServletWrapper</servlet-class>
+	</servlet>
+
+	<servlet-mapping>
+	<servlet-name>PortletWrapper</servlet-name>
+	<url-pattern>/PortletWrapper</url-pattern>
+	</servlet-mapping>
+
+	<taglib>
+	<taglib-uri>http://java.sun.com/portlet</taglib-uri>
+	<taglib-location>/WEB-INF/tld/portlet.tld</taglib-location>
+	</taglib>
+</web-app>
\ No newline at end of file
diff --git a/tuna/Jsr268EventPublisherPortlet/target/Jsr268EventPublisherPortlet-1.0-SNAPSHOT/xhtml/help.jsp b/tuna/Jsr268EventPublisherPortlet/target/Jsr268EventPublisherPortlet-1.0-SNAPSHOT/xhtml/help.jsp
new file mode 100644
index 0000000..276b480
--- /dev/null
+++ b/tuna/Jsr268EventPublisherPortlet/target/Jsr268EventPublisherPortlet-1.0-SNAPSHOT/xhtml/help.jsp
@@ -0,0 +1,3 @@
+<%@ taglib uri="http://java.sun.com/jstl/core" prefix="c" %>
+
+<h4>Need help?</h4>
\ No newline at end of file
diff --git a/tuna/Jsr268EventPublisherPortlet/target/Jsr268EventPublisherPortlet-1.0-SNAPSHOT/xhtml/view.jsp b/tuna/Jsr268EventPublisherPortlet/target/Jsr268EventPublisherPortlet-1.0-SNAPSHOT/xhtml/view.jsp
new file mode 100644
index 0000000..8c001f0
--- /dev/null
+++ b/tuna/Jsr268EventPublisherPortlet/target/Jsr268EventPublisherPortlet-1.0-SNAPSHOT/xhtml/view.jsp
@@ -0,0 +1,32 @@
+<%@ taglib uri="http://java.sun.com/portlet_2_0" prefix="portlet"%>
+<%@ taglib uri="http://java.sun.com/jstl/core" prefix="c" %>
+<portlet:defineObjects/>
+ 
+ <div>
+    <p>Use the form below to create a contact. The data is sent over to other portlets.</p>
+    <%-- The action name must be passed as a param, not as an actionURL attribute. Fix will come with Liferay 5.0.2 --%>
+    <form method="post" action="<portlet:actionURL>
+    
+    <portlet:param name="javax.portlet.action" value="savecontact" />
+    </portlet:actionURL>">
+    <%-- define savecontact ACTION, this action is processed by ProcessAction savecontact --%>
+    
+    <fieldset>
+    <table>
+    <tbody>
+    	<tr>
+    	<th><label>Contact name:</label></th>
+        <td><input name="name" /></td>
+        </tr>
+        <tr>
+        <th><label>Contact e-mail:</label></th>
+        <td><input name="email" /></td>
+        </tr>
+        <tr>
+        <td colspan="2"><input type="submit" value="Save contact" /></td>
+        </tr>
+        </tbody>
+        </table>
+    </fieldset>
+   </form>
+</div>
\ No newline at end of file
diff --git a/tuna/Jsr268EventPublisherPortlet/target/maven-archiver/pom.properties b/tuna/Jsr268EventPublisherPortlet/target/maven-archiver/pom.properties
new file mode 100644
index 0000000..9f38c2c
--- /dev/null
+++ b/tuna/Jsr268EventPublisherPortlet/target/maven-archiver/pom.properties
@@ -0,0 +1,5 @@
+#Generated by Maven
+#Tue Sep 18 07:38:22 GMT+00:07 2012
+version=1.0-SNAPSHOT
+groupId=org.exoplatform.cs
+artifactId=Jsr268EventPublisherPortlet
diff --git a/tuna/My-JSR168-Portlet/pom.xml b/tuna/My-JSR168-Portlet/pom.xml
index 9a99f3b..0e6e4aa 100644
--- a/tuna/My-JSR168-Portlet/pom.xml
+++ b/tuna/My-JSR168-Portlet/pom.xml
@@ -26,7 +26,7 @@
   </profiles>
   <dependencies>
       <dependency>
-          <groupId>portlet-api</groupId>
+          <groupId>javax.portlet</groupId>
           <artifactId>portlet-api</artifactId>
           <version>1.0</version>
           <scope>provided</scope>
